layers/+lang/python/packages.el

```diff
@@ -454,9 +454,13 @@
   (when (configuration-layer/package-used-p 'anaconda-mode)
     (add-hook 'python-mode-hook
               'spacemacs//disable-semantic-idle-summary-mode t))
-  (spacemacs/add-to-hook 'python-mode-hook
-                         '(semantic-mode
-                           spacemacs//python-imenu-create-index-use-semantic-maybe)))
+  (add-hook 'python-mode-hook
+            #'(lambda ()
+                (setq-local imenu-create-index-function-orig
+                            imenu-create-index-function)
+                (semantic-mode)))
+  (add-hook 'semantic-mode-hook
+            'spacemacs//python-imenu-create-index-use-semantic-maybe))
```

Comment:
`semantic-mode` is a global mode so this might now run in non-python buffers which is a bug.  Instead this should probably just be on python-mode-hook.